rpcserver.go

```diff
@@ -6086,67 +6129,77 @@ func (s *rpcServer) jsonRPCRead(w http.ResponseWriter, r *http.Request, isAdmin
 	conn.SetReadDeadline(timeZeroVal)
 
 	// Attempt to parse the raw body into a JSON-RPC request.
-	var responseID interface{}
 	var jsonErr error
-	var result interface{}
-	var request dcrjson.Request
+	var msg []byte
+	var request interface{}
 	if err := json.Unmarshal(body, &request); err != nil {
 		jsonErr = &dcrjson.RPCError{
 			Code: dcrjson.ErrRPCParse.Code,
 			Message: fmt.Sprintf("Failed to parse request: %v",
 				err),
 		}
-	}
-	if jsonErr == nil {
-		// Requests with no ID (notifications) must not have a response
-		// per the JSON-RPC spec.
-		if request.ID == nil {
-			return
-		}
 
-		// The parse was at least successful enough to have an ID so
-		// set it for the response.
-		responseID = request.ID
+		msg, err = createMarshalledReply("", "", jsonErr)
+		if err != nil {
+			rpcsLog.Errorf("Failed to marshal reply: %v", err)
+		}
+	}
 
+	if jsonErr == nil {
 		// Setup a close notifier.  Since the connection is hijacked,
 		// the CloseNotifer on the ResponseWriter is not available.
 		closeChan := make(chan struct{}, 1)
 		go func() {
-			_, err := conn.Read(make([]byte, 1))
+			_, err = conn.Read(make([]byte, 1))
 			if err != nil {
 				close(closeChan)
 			}
 		}()
 
-		// Check if the user is limited and set error if method
-		// unauthorized
-		if !isAdmin {
-			if _, ok := rpcLimited[request.Method]; !ok {
-				jsonErr = rpcInvalidError("limited user not " +
-					"authorized for this method")
+		// determine if the request is single or batched
+		switch request.(type) {
+		case map[string]interface{}:
+			{
```

Comment:
can you drop the scope and save on indentation?

Reply:
sure thing, on it.